src/Npgsql/NpgsqlCommand.cs

```diff
@@ -1248,6 +1248,11 @@ internal async ValueTask<NpgsqlDataReader> ExecuteReader(CommandBehavior behavio
 
                         connector.UserTimeout = CommandTimeout * 1000;
 
+                        // If statement logging is enabled, start a StopWatch to log the query's duration time.
```

Comment:
This can all move into the LogCommand method that's already called above, when debug logging is enabled. This would executing the same condition twice (loglevel check), and wouldn't allocate the Stopwatch on the heap unless needed.

In addition, since Stopwatch is class and so allocated on the heap, we shouldn't be allocating a new one every time a command is executed. We can have a single Stopwatch instance on the connector and reuse it many times.